CI failing for docker proxy tests using docker-py

## What you expected to happen?

CI to be passing

## What happened?

tests (test/*proxy_docker_py_test.sh) that verify weave docker API proxy functionliaty are failing. I could not trace root cause for tests failing. Updating current `docker-py` image  used `weaveworks/docker-py:1.9.0-prime` to recent versions of `joffrey/docker-py` did not help either. Failing tests vary across the versions of docker-py but could not get clean test run.

```
tests/integration/network_test.py .....F.s.....F...
=========================== short test summary info ============================
SKIP [1] tests/integration/network_test.py:363: API version is too low (< 1.23)

=================================== FAILURES ===================================
_____________________ TestNetworks.test_connect_with_links _____________________
tests/integration/network_test.py:306: in test_connect_with_links
    self.execute(container, ['nslookup', 'bar'])
tests/helpers.py:185: in execute
    assert actual_exit_code == exit_code, msg
E   AssertionError: Expected `nslookup bar` to exit with code 0 but returned 1:
E   Server:		127.0.0.11
E   Address:	127.0.0.11:53
E   
E   ** server can't find bar.c.weave-net-tests.internal: NXDOMAIN
E   
E   *** Can't find bar.google.internal: No answer
E   *** Can't find bar.c.weave-net-tests.internal: No answer
E   *** Can't find bar.google.internal: No answer
_____________________ TestNetworks.test_create_with_links ______________________
tests/integration/network_test.py:275: in test_create_with_links
    self.execute(container, ['nslookup', 'bar'])
tests/helpers.py:185: in execute
    assert actual_exit_code == exit_code, msg
E   AssertionError: Expected `nslookup bar` to exit with code 0 but returned 1:
E   Server:		127.0.0.11
E   Address:	127.0.0.11:53
E   
E   ** server can't find bar.c.weave-net-tests.internal: NXDOMAIN
E   
E   *** Can't find bar.google.internal: No answer
E   *** Can't find bar.c.weave-net-tests.internal: No answer
E   *** Can't find bar.google.internal: No answer
=============== 2 failed, 21 passed, 1 skipped in 30.79 seconds ================
test #1 "false" failed:
	program terminated with code 1 instead of 0
1 of 2 tests failed in 69.887s.
```

## How to reproduce it?

CI run from [12055](https://circleci.com/gh/weaveworks/weave/12055) are failing with same issue.

--- a/test/600_proxy_docker_py.sh
+++ b/test/600_proxy_docker_py.sh
@@ -13,6 +13,9 @@ docker_py_test() {
 
     # Work round https://github.com/docker/docker-py/issues/852
     docker_on $HOST1 pull busybox:buildroot-2014.02 >/dev/null
+    # Work round https://github.com/weaveworks/weave/issues/3680 - pin busybox version to 1.28
+    docker_on $HOST1 pull busybox:1.28 >/dev/null
+    docker_on $HOST1 tag busybox:1.28 busybox:latest >/dev/null
     # Get a list of the tests for use to shard
     docker_on $HOST1 pull $IMAGE >/dev/null
     CANDIDATES=$(docker_on $HOST1 run \

